server/hedley/modules/custom/hedley_reports/hedley_reports.module

```diff
@@ -5674,193 +5680,3072 @@ function hedley_reports_generate_completion_data_for_tuberculosis($participant,
+              $question['value'],
+              ['dizziness', 'low-urine-output', 'dark-urine']
+            )) {
+              $question_recorded = TRUE;
+              break;
+            }
+          }
+
+          if (!$question_recorded) {
+            $activity_expected = !hedley_reports_prenatal_symptom_recorded_previously($encounter_data, 'nausea-and-vomiting');
+          }
+        }
+      }
+    }
+  }
+
+  // Education for Leg Cramps, Lower Back Pain, Constipation and
+  // Varicose Veins symptoms.
+  if (!$activity_expected) {
+    if (empty($symptom_review_measurement) || empty($symptom_review_measurement->field_prenatal_symptoms[LANGUAGE_NONE])) {
+      foreach ($symptom_review_measurement->field_prenatal_symptoms[LANGUAGE_NONE] as $symptom) {
+        if (in_array(
+          $symptom['value'],
+          ['leg-cramps', 'low-back-pain', 'constipation', 'varicose-veins']
+        )) {
+          $activity_expected = TRUE;
+          break;
+        }
+      }
+    }
+  }
+
+  // Leg pain / redness education.
+  if (!$activity_expected) {
+    // We need to make sure that leg-pain-redness symptom is recorded, and
+    // neither of follow-up questions (leg-painful, leg-swollen and leg-warm)
+    // are answered positively.
+    // Then, we make sure that Nausea and Vomiting symptom was not recorded
+    // during previous encounters.
+    if (empty($symptom_review_measurement) || empty($symptom_review_measurement->field_prenatal_symptoms[LANGUAGE_NONE])) {
+      $symptom_recorded = FALSE;
+      foreach ($symptom_review_measurement->field_prenatal_symptoms[LANGUAGE_NONE] as $symptom) {
+        if ($symptom['value'] === 'leg-pain-redness') {
+          $symptom_recorded = TRUE;
+          break;
+        }
+      }
+
+      if ($symptom_recorded) {
+        $symptom_questions = $symptom_review_measurement->field_prenatal_symptom_questions[LANGUAGE_NONE];
+        if (!empty($symptom_questions)) {
+          $question_recorded = FALSE;
+          foreach ($symptom_questions as $question) {
+            if (in_array(
+              $question['value'],
+              ['leg-painful', 'leg-swollen', 'leg-warm']
+            )) {
+              $question_recorded = TRUE;
+              break;
+            }
+          }
+
+          $activity_expected = !$question_recorded;
+        }
+      }
+    }
+  }
+
+  // Pelvic pain education.
+  if (!$activity_expected) {
+    // First, checking by current encounter.
+    // We need to make sure that pelvic-pain symptom is recorded, and
+    // follow-up question (pelvic-pain-hospitalization) is answered negatively.
+    // Then, we make sure that pelvic-pain symptom was not recorded during
+    // previous encounters.
+    if (empty($symptom_review_measurement) || empty($symptom_review_measurement->field_prenatal_symptoms[LANGUAGE_NONE])) {
+      $symptom_recorded = FALSE;
+      foreach ($symptom_review_measurement->field_prenatal_symptoms[LANGUAGE_NONE] as $symptom) {
```

Comment:
We just check if it's empty and then, we try to foreach over it, is it correct?

Reply:
Yep. Good catch!  Fixed.